### PR TITLE
Added bias_and_explainability to SageMakerClarifyProcessor

### PR DESCRIPTION
**Pull request by dosatos**
*Issue #, if available:*

*Description of changes:*
Added a public method to run both Bias and Explainability in a single job processing

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/clarify.py
Comment (keerthanvasist) [inline src/sagemaker/clarify.py:1311]: Can we see if we can override the existing run() method for this purpose?

Comment (dosatos) [reply src/sagemaker/clarify.py:1371]: `Signature of method 'SageMakerClarifyProcessor.run()' does not match signature of the base method in class 'Processor'`

**Comment by dosatos** [comment]
I will apply formatting after the PR #3271 gets merged - it partially improves the style already

File: src/sagemaker/clarify.py
Comment (larroy) [inline src/sagemaker/clarify.py:1534]: I'm not a big fan of these Union types, I see the intention, but is it worth it to save the user typing [] ?  I would prefer if it's a plain list and we check that is not empty. Open to discuss about this and be convinced of the opposite though.

Comment (dosatos) [reply src/sagemaker/clarify.py:1534]: Are you saying that, instead of making the user to pass `"all"`, we can make it `["all"]`?
Actually, I did it this way to be consistent with the current implementation - i.e. with how we expose methods yet.

File: src/sagemaker/clarify.py
Comment (dosatos) [reply src/sagemaker/clarify.py:1706]: then the last line returns it, no? `return explainability_config.get_explainability_config()`

File: src/sagemaker/clarify.py
Comment (larroy) [reply src/sagemaker/clarify.py:1534]: I mean         explainability_config: Union[ExplainabilityConfig, List[ExplainabilityConfig]],

**Comment by shreyapandit** [comment]
I have restarted PR and Local Mode tests, notebook failures are expected and I can override them. Please have 1 approval from @larroy or @keerthanvasist for us to be able to merge

File: src/sagemaker/clarify.py
Comment (keerthanvasist) [reply src/sagemaker/clarify.py:1534]: So it was initially only `ExplainabilityConfig`, and we only had SHAPConfig. But then we added PDP, and we wanted to let customers be allowed to choose both of them in a single job. So, we made into an array but kept the original single ExplainabilityConfig as well to maintain backward compatibility.

Comment (larroy) [reply src/sagemaker/clarify.py:1533]: I see this was like that and you just annotated the same type, thanks for catching up with this.

**Review by larroy — APPROVED** [review]

File: src/sagemaker/clarify.py
Comment (keerthanvasist) [inline src/sagemaker/clarify.py:1551]: Just to maintain consistency, can you use `self` instead of `cls`?

**Review by keerthanvasist — APPROVED** [review]

**Comment by dosatos** [comment]
The changes in this PR were also present in this one: https://github.com/aws/sagemaker-python-sdk/pull/3296

Thus, cancelling the PR

File: src/sagemaker/clarify.py
Comment (dosatos) [reply src/sagemaker/clarify.py:1551]: Thanks for the comment, Keerthan.`self` implies creation of an instance, `cls` does not require an instance
This should also be driven by [pep8](https://stackoverflow.com/questions/4613000/difference-between-cls-and-self-in-python-classes).

**Pull request closed without merging**
